QgisModelBaker/libqgsprojectgen/dbconnector/mssql_connector.py

```diff
@@ -0,0 +1,421 @@
+                alias_left_join = """
+                    LEFT JOIN        {}.T_ILI2DB_TABLE_PROP as alias
+                        on    alias.tablename = tbls.TABLE_NAME
+                        AND alias.tag = 'ch.ehi.ili2db.dispName'
+                               """.format(self.schema)
+                model_where = """LEFT JOIN {}.t_ili2db_classname c
+                      ON tbls.TABLE_NAME = c.sqlname""".format(self.schema)
+                      
+            schema_where = "AND tbls.TABLE_SCHEMA = '{}'".format(self.schema)
+
+            cur = self.conn.cursor()
+
+            query = """
+                SELECT distinct
+                    tbls.TABLE_SCHEMA AS schemaname,
+                    tbls.TABLE_NAME AS tablename, 
+                    Col.Column_Name AS primary_key,
+                    clm.COLUMN_NAME AS geometry_column,
+                    tsrid.setting as srid,
+                    {kind_settings_field}
+                    {table_alias}
+                    {model_name}
+                    {ili_name}
+                    {extent}
+                    tgeomtype.setting as simple_type,
+                    null as formatted_type
+                FROM
+                INFORMATION_SCHEMA.TABLE_CONSTRAINTS Tab 
+                INNER JOIN INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE Col 
+                    on Col.Constraint_Name = Tab.Constraint_Name
+                    AND Col.Table_Name = Tab.Table_Name
+                    AND Col.CONSTRAINT_SCHEMA = Tab.CONSTRAINT_SCHEMA
+                RIGHT JOIN INFORMATION_SCHEMA.TABLES as tbls
+                    on Tab.TABLE_NAME = tbls.TABLE_NAME 
+                    AND Tab.CONSTRAINT_SCHEMA = tbls.TABLE_SCHEMA
+                    AND Tab.Constraint_Type = 'PRIMARY KEY'
+                {domain_left_join}
+                {alias_left_join}
+                {model_where}
+                LEFT JOIN INFORMATION_SCHEMA.COLUMNS as clm
+                    on clm.TABLE_NAME = tbls.TABLE_NAME
+                    AND clm.TABLE_SCHEMA = tbls.TABLE_SCHEMA
+                    AND clm.DATA_TYPE = 'geometry'
+                LEFT JOIN {schema}.T_ILI2DB_COLUMN_PROP as tsrid
+                    on tbls.TABLE_NAME = tsrid.tablename and clm.COLUMN_NAME = tsrid.columnname and tsrid.tag='ch.ehi.ili2db.srid'
+                LEFT JOIN {schema}.T_ILI2DB_COLUMN_PROP as tgeomtype
+                    on tbls.TABLE_NAME = tgeomtype.tablename and clm.COLUMN_NAME = tgeomtype.columnname and tgeomtype.tag= 'ch.ehi.ili2db.geomType' 
+                WHERE tbls.TABLE_TYPE = 'BASE TABLE' {schema_where}
+            """.format(kind_settings_field=kind_settings_field, table_alias=table_alias, model_name=model_name, ili_name=ili_name, extent=extent,
+                    domain_left_join=domain_left_join, alias_left_join=alias_left_join, model_where=model_where,
+                    schema_where=schema_where,schema=self.schema)
+            cur.execute(query)
```

Comment:
as far as I see (did not check properly), this does look like the same query than in PG. Could this be move in the parent class to avoid code duplication?

Reply:
We expect the same response structure, but SQLs are specific for each DB engine.